src/main/java/org/folio/innreach/domain/service/impl/InnReachTransactionServiceImpl.java

```diff
@@ -0,0 +1,49 @@
+package org.folio.innreach.domain.service.impl;
+
+import javax.persistence.EntityExistsException;
+import lombok.RequiredArgsConstructor;
+import org.folio.innreach.domain.entity.InnReachTransaction;
+import org.folio.innreach.domain.entity.InnReachTransaction.TransactionType;
+import org.folio.innreach.domain.service.CentralServerService;
+import org.folio.innreach.domain.service.InnReachTransactionService;
+import org.folio.innreach.dto.D2irResponseDTO;
+import org.folio.innreach.dto.TransactionItemHoldDTO;
+import org.folio.innreach.mapper.InnReachTransactionMapper;
+import org.folio.innreach.repository.InnReachTransactionRepository;
+import org.springframework.stereotype.Service;
+
+@RequiredArgsConstructor
+@Service
+public class InnReachTransactionServiceImpl implements InnReachTransactionService {
+
+  private final InnReachTransactionRepository repository;
+  private final InnReachTransactionMapper mapper;
+  private final CentralServerService centralServerService;
+
+  @Override
+  public D2irResponseDTO createInnReachTransactionItemHold(String trackingId, String centralCode, TransactionItemHoldDTO dto) {
+    var response = new D2irResponseDTO();
+    response.setStatus("ok");
+    try {
+      centralServerService.getCentralServerByCentralCode(centralCode);
+      InnReachTransaction transaction = new InnReachTransaction();
+      transaction.setTrackingId(trackingId);
+      transaction.setCentralServerCode(centralCode);
+      transaction.setType(TransactionType.ITEM);
+      transaction.setState(InnReachTransaction.TransactionState.ITEM_HOLD);
+      var itemHold = mapper.toItemHold(dto);
+      transaction.setHold(itemHold);
+      repository.fetchOneByTrackingId(trackingId).ifPresent(t -> {
```

Comment:
As far as I remember, we should have DB constraints for such cases, and we shouldn't do checks like this manually